pkg/reconciler/apiserversource/apiserversource.go

```diff
@@ -392,34 +393,38 @@ func (r *Reconciler) getLabelSelector(src *v1alpha1.ApiServerSource) labels.Sele
 	return labels.SelectorFromSet(resources.Labels(src.Name))
 }
 
-func (r *Reconciler) updateStatus(ctx context.Context, desired *v1alpha1.ApiServerSource) (*v1alpha1.ApiServerSource, error) {
-	apiserversource, err := r.apiserversourceLister.ApiServerSources(desired.Namespace).Get(desired.Name)
-	if err != nil {
-		return nil, err
-	}
+func (r *Reconciler) updateStatus(ctx context.Context, original, desired *v1alpha1.ApiServerSource) error {
+	existing := original.DeepCopy()
```

Comment:
if there is an update to the version in the api server, this will just error and error, that was what the original Get was for.

Reply:
the original code Gets the current copy and then merges with the updated status and the current spec

Reply:
thanks for your review @n3wscott. Does this mean that https://github.com/knative/serving/pull/6162 and https://github.com/google/knative-gcp/pull/508 are also problematic and should be rolled back? fyi @markusthoemmes @nachocano

Reply:
Sorry, I deleted my comment by accident. Thanks for the review. So this PR is mostly copying work done in https://github.com/knative/serving/pull/6162 and https://github.com/google/knative-gcp/pull/508. Does this mean that those PRs are also problematic and should be rolled back? fyi @markusthoemmes @nachocano